### PR TITLE
MXF parsing: terminate strings on the first null character

### LINKED ISSUE #372: Handle MXF string termination to avoid incorrect validation errors
MXF Header Metadata string properties that contain a null character should be trimmed. Currently, the terminating null character (and any subsequent characters) are included in the string used by Photon for validation, resulting in incorrect error reports such as:

```
[main] ERROR com.netflix.imflibrary.app.IMPAnalyzer -           ERROR-Language Code ([0x656e2d474200, en-GB.]) in SoundFieldGroupLabelSubdescriptor in the IMFTrackfile represented by ID ac4aaec3-a55c-48d0-a806-6cb92888ab3a is not RFC5646 compliant [Photon version: 4.10.0-SNAPSHOT]
```

### PR DESCRIPTION
**Pull request by thomasheritage**
Per SMPTE ST 377-1:2019, a "zero value" can be used to terminate a string.

So, discard any data after the first null character found (if any).

This has the same effect as the implementation for strings in regxmllib at: https://github.com/sandflow/regxmllib/blob/b47b0cdbd6dbf51cae4fb14c3a6a827eb3e0e2cc/src/main/java/com/sandflow/smpte/regxml/FragmentBuilder.java#L902

Note that regxmllib also uses the same `readCharacters` method for MXF properties with a Type of Type Kind "Character", and allows values of this Type to contain the null character.
Photon does not support such "Character" Types (and indeed none have ever been used in MXF according to the SMPTE Metadata Registers). So, no consideration is given to these Types in this patch.

Closes #372

### CODE REVIEW

File: src/main/java/com/netflix/imflibrary/MXFPropertyPopulator.java [hunk 1]
```diff
@@ -240,15 +240,15 @@ else if ((field.getType() == Boolean.class) && (byteArraySize == 1))
...
+     * Getter for a string representing the byte[], terminating on the first null character
      *
      * @param byteArray the byte array
      * @param charset the charset
      * @return the string
      */
     public static String getString(byte[] byteArray, Charset charset)
     {
-        return new String(byteArray, charset);
+        return new String(byteArray, charset).split("\u0000", 2)[0];
```
Comment (palemieux) [inline src/main/java/com/netflix/imflibrary/MXFPropertyPopulator.java:251]: I am not in love with the `split`.

What about?

```
         String s = String(byteArray, charset);

         int i = s.indexOf('\u0000');

        if (i== -1) {
            return s;
        }

        return s.substring(0, i);
```

**Review by palemieux — CHANGES_REQUESTED** [review]
A test case should ideally be added.

**Comment by thomasheritage** [comment]
I've updated the branch to amend the code and add a test (with a small MXF file containing Header Metadata with strings that are null terminated: prior to the code fix an error was given by Photon for this file per https://github.com/Netflix/photon/issues/372#issue-2375490069 )

**Review by palemieux — APPROVED** [review]
